go.mod

```diff
@@ -3,23 +3,23 @@ module github.com/SovereignCloudStack/cluster-stack-operator
 go 1.21
 
 require (
-	github.com/go-logr/logr v1.3.0
+	github.com/go-logr/logr v1.4.1
 	github.com/go-task/slim-sprig v0.0.0-20230315185526-52ccab3ef572
 	github.com/google/go-github/v52 v52.0.0
-	github.com/onsi/ginkgo/v2 v2.13.1
-	github.com/onsi/gomega v1.30.0
-	github.com/stretchr/testify v1.8.4
-	golang.org/x/mod v0.14.0
-	golang.org/x/oauth2 v0.14.0
-	k8s.io/api v0.28.4
-	k8s.io/apimachinery v0.28.4
-	k8s.io/cli-runtime v0.28.4
-	k8s.io/client-go v0.28.4
-	k8s.io/klog/v2 v2.100.1
+	github.com/onsi/ginkgo/v2 v2.17.0
+	github.com/onsi/gomega v1.32.0
+	github.com/stretchr/testify v1.9.0
+	golang.org/x/mod v0.16.0
+	golang.org/x/oauth2 v0.18.0
+	k8s.io/api v0.29.3
+	k8s.io/apimachinery v0.29.3
+	k8s.io/cli-runtime v0.29.3
+	k8s.io/client-go v0.29.3
+	k8s.io/klog/v2 v2.120.1
```

Comment:
```json5
{
  description: "Disable update k8s packages",
  matchManagers: ["gomod"],
  matchDepNames: ["k8s.io/api", "k8s.io/apimachinery", "k8s.io/apiserver", "k8s.io/client-go", "k8s.io/kubectl", "k8s.io/code-generator"],
  matchUpdateTypes: ["patch"],
  enabled: true,
}
```
Even though we have disabled it, it's getting bumped, I think we need to update this list with some more deps that are bumping this to 0.29.3 

For example, I see `k8s.io/apiextensions-apiserver` and `k8s.io/component-base` missing from the list. 

// cc @janiskemper